recipes/sphinx-releases/meta.yaml

```diff
@@ -0,0 +1,43 @@
+{% set name = "sphinx-releases" %}
```

Comment:
Can understand the value of disambiguation here. Though `releases` is the PyPI name. Thoughts?

cc @conda-forge/staged-recipes

Reply:
`releases` is an unfortunate name. Other conda sphinx plugins use the `sphinx-` prefix so I thought it best to follow suit. I don't have strong opinions either way.